vendor/sigs.k8s.io/kubesdk/pkg/resource/manager/gcp/redis/manager.go

```diff
@@ -188,7 +195,7 @@ func (rm *RsrcManager) Observe(observables ...resource.Observable) (*resource.Ba
 func (rm *RsrcManager) Update(item resource.Item) error {
 	obj := item.Obj.(*Object)
 	d := obj.Obj
-	_, err := rm.service.Projects.Locations.Instances.Patch(d.Name, d).
+	_, err := rm.service.Projects.Locations.Instances.Patch(obj.Parent + "/instances/" + obj.InstanceID, d).
```

Comment:
What is the impact of this. Hadn't considered this earlier.

Reply:
the `Name` field of a MemoryStore instance would be automatically created with this format: `projects/project-id/locations/location-id/instances/instance-name`, when we update the spec of a running MemoryStore, the `Name` field of this `item` is an empty string (since we took it out of `MemoryStoreSpec`) which will raise a 404 error.